lib/utilities/clean-remove.js

```diff
@@ -4,45 +4,36 @@ const path = require('path');
 const fs = require('fs-extra');
 const walkUp = require('./walk-up-path');
 
-function cleanRemove(fileInfo) {
-  // see if file exists to avoid wasting time
-  return fs
-    .stat(fileInfo.outputPath)
-    .then(() => fs.remove(fileInfo.outputPath))
-    .then(() => {
-      let paths = walkUp(fileInfo.displayPath).map((thePath) => path.join(fileInfo.outputBasePath, thePath));
+async function cleanRemove(fileInfo) {
+  try {
+    await fs.stat(fileInfo.outputPath);
+    await fs.remove(fileInfo.outputPath);
+    let paths = walkUp(fileInfo.displayPath).map((thePath) => path.join(fileInfo.outputBasePath, thePath));
 
-      return paths.reduce(
-        (chainedPromise, thePath) =>
-          chainedPromise.then((wasShortCircuited) => {
-            if (wasShortCircuited) {
-              // optimization that says since my child dir wasn't empty,
-              // I can't be empty, so keep skipping
-              return true;
-            }
+    for (let thePath of paths) {
+      try {
+        let childPaths = await fs.readdir(thePath);
+        if (childPaths.length > 0) {
+          return;
+        }
 
-            // get list of files remaining in this dir
-            return fs.readdir(thePath).then((paths) => {
-              if (paths.length) {
-                // don't check parent dirs since this one isn't empty
-                return true;
-              }
+        await fs.remove(thePath);
+      } catch (err) {
+        if (err.code === 'ENOENT') {
+          // do nothing, already removed
+        }
```

Comment:
I'm not sure that both this `catch` and the outer one are needed.

Can you remove this try/catch (and we'll just rely on the outer one)?

Reply:
Done :)